src/uu/split/src/split.rs

```diff
@@ -65,6 +67,86 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
     }
 }
 
+/// If the same separator (with the same value) was used multiple times - `split` should NOT fail
+/// If the separator was used multiple times but with different values - `split` should fail
+/// Unfortunately Clap cannot support both conditions simultaneously, so it is setup to fail on the second condition
+/// For the first condition - this function will find if there are multiple separator options with the same value
+/// and remove duplicates, leaving only one
+/// i.e. `split -t: -t:` will become `split -t:`
+/// However it will leave other values untouched, so
+/// `split -t: -t: -t,` will become `split -t: -t,` and Clap will panic next as it should
+fn handle_multiple_separator_options(args: &[OsString]) -> Vec<OsString> {
+    let mut separator_values: Vec<&str> = vec![];
+    let mut preceding_sep_opt_req_value = false;
+    let mut i = -1;
+    let mut slices_to_remove: Vec<i32> = vec![];
+
+    // collect slices to remove
+    for os_slice in args {
+        i += 1;
+        if let Some(slice) = os_slice.to_str() {
```

Comment:
this feels quite complex, maybe you could do something like:

         let (prefix, offset) = if slice.starts_with("-t") {
                ("-t", 2)
            } else if slice.starts_with("--separator=") {
                ("--separator=", 12)
            } else if slice.starts_with("--separator") {
                ("--separator", 11)
            } else {
                ("", 0)
            };

Reply:
this does not take into account concatenated value vs value separated by whitespace, but I think it can be added into this format for each if/if else. Will try to refactor.
thanks!

Reply:
unfortunately it does not look like it can be much simplified here, since both cases need to be covered - concatenated values like `--separator=:` or `-t,` as well as values separated by white space like `--separator :` or `-t ,` 
We cannot just get (prefix, offset) - it would only work for concatenated case.

Reply:
Btw - this function exists only to pass GNU test for split separator